packages/core/src/report.ts

```diff
@@ -223,25 +243,180 @@ export class AllureReport {
+        continue;
+      }
+
+      if (content instanceof PathResultFile) {
+        await addEntry(createReadStream(content.path), {
+          name: attachment.id,
+        });
+      } else {
+        await addEntry(await content.asBuffer(), {
+          name: attachment.id,
+        });
+      }
+    }
+
+    dumpArchive.finalize();
+
+    return promise as Promise<void>;
+  };
+
+  restoreState = async (stages: string[]): Promise<void> => {
+    for (const stage of stages) {
+      if (!existsSync(stage)) {
+        continue;
+      }
+
+      const dump = new ZipReadStream.async({
+        file: stage,
+      });
+
+      const testResultsEntry = await dump.entryData(AllureStoreDumpFiles.TestResults);
+      const testCasesEntry = await dump.entryData(AllureStoreDumpFiles.TestCases);
+      const fixturesEntry = await dump.entryData(AllureStoreDumpFiles.Fixtures);
+      const attachmentsEntry = await dump.entryData(AllureStoreDumpFiles.Attachments);
+      const environmentsEntry = await dump.entryData(AllureStoreDumpFiles.Environments);
+      const reportVariablesEntry = await dump.entryData(AllureStoreDumpFiles.ReportVariables);
+      const globalAttachmentsEntry = await dump.entryData(AllureStoreDumpFiles.GlobalAttachments);
+      const globalErrorsEntry = await dump.entryData(AllureStoreDumpFiles.GlobalErrors);
+      const indexAttachmentsEntry = await dump.entryData(AllureStoreDumpFiles.IndexAttachmentsByTestResults);
+      const attachmentsEntries = Object.entries(await dump.entries()).reduce((acc, [entryName, entry]) => {
+        switch (entryName) {
+          case AllureStoreDumpFiles.Attachments:
+          case AllureStoreDumpFiles.TestResults:
+          case AllureStoreDumpFiles.TestCases:
+          case AllureStoreDumpFiles.Fixtures:
+          case AllureStoreDumpFiles.Environments:
+          case AllureStoreDumpFiles.ReportVariables:
+          case AllureStoreDumpFiles.GlobalAttachments:
+          case AllureStoreDumpFiles.GlobalErrors:
+          case AllureStoreDumpFiles.IndexAttachmentsByTestResults:
+            return acc;
+          default:
+            return Object.assign(acc, {
+              [entryName]: entry,
+            });
+        }
+      }, {});
+      const dumpState: AllureStoreDump = {
+        testResults: JSON.parse(testResultsEntry.toString("utf8")),
+        testCases: JSON.parse(testCasesEntry.toString("utf8")),
+        fixtures: JSON.parse(fixturesEntry.toString("utf8")),
+        attachments: JSON.parse(attachmentsEntry.toString("utf8")),
+        environments: JSON.parse(environmentsEntry.toString("utf8")),
+        reportVariables: JSON.parse(reportVariablesEntry.toString("utf8")),
+        globalAttachments: JSON.parse(globalAttachmentsEntry.toString("utf8")),
+        globalErrors: JSON.parse(globalErrorsEntry.toString("utf8")),
+        indexAttachmentByTestResult: JSON.parse(indexAttachmentsEntry.toString("utf8")),
+      };
+      const stageTempDir = join(tmpdir(), stage);
```

Comment:
Here, `stage` is an absolute path to the stage dump. Joining it with `tmpdir()` results in an invalid path like `C:\Users\username\AppData\Local\Temp\C:\Users\username\repos\allure-framework\allure3\s3.zip`.

Either extract the stage name from this path or use `mkdtemp` as I suggested in my other comment.